code/game/objects/structures/tables_racks.dm

```diff
@@ -583,6 +599,57 @@
 	return ..()
 
 /*
+ * Glass
+ */
+
+/obj/structure/table/glass
+	name = "glass table"
+	desc = "A standard table with a fine glass finish."
+	icon_state = "glass_table"
+	parts = /obj/item/weapon/table_parts/glass
+	health = 30
+
+/obj/structure/table/glass/attackby(obj/item/W as obj, mob/user as mob, params)
+	if (istype(W, /obj/item/weapon/grab) && get_dist(src,user)<2)
+		var/obj/item/weapon/grab/G = W
+		if (istype(G.affecting, /mob/living))
+			var/mob/living/M = G.affecting
+			if (G.state < GRAB_AGGRESSIVE)
+				if(user.a_intent == I_HURT)
+					if (prob(15))	M.Weaken(5)
+					M.apply_damage(15,def_zone = "head")
+					visible_message("<span class='warning'>[G.assailant] slams [G.affecting]'s face against \the [src]!</span>")
+					playsound(get_turf(src), 'sound/weapons/tablehit1.ogg', 50, 1)
+					playsound(src.loc, "shatter", 50, 1) //WRESTLEMANIA tax
+					new /obj/item/weapon/shard(src.loc)
+					new /obj/item/weapon/shard(src.loc)
```

Comment:
Interesting how breaking a glass table with kicks makes 1 shard and table parts but throwing somebody ONTO it produces 2 shards.